src/masternode-payments.cpp

```diff
@@ -777,6 +779,62 @@ bool CMasternodePayments::ProcessBlock(int nBlockHeight, CConnman& connman)
     return false;
 }
 
+void CMasternodePayments::CheckPreviousBlockVotes(int nPrevBlockHeight)
+{
+    LogPrint("mnpayments", "CMasternodePayments::CheckPreviousBlockVotes -- nPrevBlockHeight=%d, expected voting MNs:\n",
```

Comment:
... and it should also only run if we are in (specific) debug mode so
```
if (!LogAcceptCategory("mnpayments")) return;
```
can save some work too.
And then you could also replace `LogPrint`-s inside the function with `LogPrintf`-s because there will be no more need to check the category (however... see below).

Reply:
This is the way I had it in the beginning, but then realized that it would also skip updating of the mapMasternodesDidNotVote map. This will be inconvenient in case you decide to enable the debug category when the node was already running for some time.

Reply:
Hmm, true. But on the other hand - we don't use `mapMasternodesDidNotVote` for anything else, it's for debugging purposes only and still adds some overhead even when it's not used. But the overhead is negligible though (I see 30-40ms on my machine), so it's probably ok.